### PR TITLE
LimitedStack benchmark & change to sliding array

### PR DESCRIPTION
**Pull request by MrHant**
LimitedStack implementation can be changed to sliding array
Benchmark shows following improvement in speed, but almost same memory wise:

image

### CODE REVIEW

File: Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
 namespace UniState
 {
     public class LimitedStack<T>
     {
-        private readonly List<T> _list;
+        private readonly  T[] _list;
+        private int _top = 0;
+        private int _bot = 0;
+        private int _maxSize;
+        // count: (_top - _bot) % _maxSize
```
Comment (bazyleu) [inline Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs:9]: Lets implement following methods and remove comments:
- int Count()
- T ToArray() (Convert stack to array => create and return new array where size of array is Count, first element of array [0] is bottom element of the stack, last element of array [^1] is the top element of the stack.

This method will be used in the future versions.

**Review by bazyleu — CHANGES_REQUESTED** [review]
@MrHant thanks for sharing result of the investigations and performance testing 👍 
Lets merge your optimization and I release next version for this improvement (0.6.1).

I would like to ask some Pull Request modifications before merge.
1. Lets remove all files except [LimitedStack.cs](https://github.com/bazyleu/UniState/pull/12/files#diff-811e6bb7c59db73bef9e316644213da66a9b5c95cd3290c721290b1fe7bd8d07) from Pull Request.
2. Lets resolve comments in [LimitedStack.cs](https://github.com/bazyleu/UniState/pull/12/files#diff-811e6bb7c59db73bef9e316644213da66a9b5c95cd3290c721290b1fe7bd8d07).

Thanks!

File: Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs [hunk 2]
```diff
@@ -1,37 +1,46 @@
-using System.Collections.Generic;
-
 namespace UniState
 {
     public class LimitedStack<T>
     {
-        private readonly List<T> _list;
+        private readonly  T[] _list;
```
[the remark this answers is not in the document]
Comment (MrHant) [reply Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs]: sure

File: Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs (hunk 1 above)
Comment (MrHant) [reply Assets/UniState/Runtime/Core/StateMachine/LimitedStack.cs]: Added.

**Comment by MrHant** [comment]
Made requested changes. Please take a look.

**Review by bazyleu — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**